src/firebaseLib.php

```diff
@@ -231,13 +231,22 @@ private function _getCurlHandler($path, $mode, $options = array())
         return $ch;
     }
 
-    private function _writeData($path, $data, $method = 'PUT', $options = array())
-    {
-        $jsonData = json_encode($data);
+    private function _writeData($path, $data, $method = 'PUT', $options = array(), $contentHeader = array())
+    {   
         $header = array(
             'Content-Type: application/json',
             'Content-Length: ' . strlen($jsonData)
         );
+
+        // Insert additional settings into the request header 
+        if(!empty($contentHeader)) {
+            foreach($contentHeader as $key => $item) {
+                $header[$key] = $item;
+            }
+        }
+
+        $jsonData = json_encode($data);
```

Comment:
The `json_encode` needs to come before you set the `Content-Length` header. The encoding process is going to lengthen the string, so your length header is going to be wrong if you don't encode first.

Reply:
good catch